tools/nemo_forced_aligner/utils/data_prep.py

```diff
@@ -299,9 +298,14 @@ def get_batch_tensors_and_boundary_info(manifest_lines_batch, model, separator,
     # and (optionally) the predicted ASR text from the hypotheses
     audio_filepaths_batch = [line["audio_filepath"] for line in manifest_lines_batch]
     B = len(audio_filepaths_batch)
-    with torch.no_grad():
-        hypotheses = model.transcribe(audio_filepaths_batch, return_hypotheses=True, batch_size=B)
-
+    if not simulate_cache_aware_streaming:
+        with torch.no_grad():
+            hypotheses = model.transcribe(audio_filepaths_batch, return_hypotheses=True, batch_size=B)
```

Comment:
The way of switching to local attention by sacrifice a little accuracy seems interesting. But for now, let's only consider the arbitrary cache aware streaming model.

Reply:
Sure, let's keep the cache-aware but as we don't we have any cache-aware checkpoints publicly available, it would be nice to give users the option of using buffered streaming with offline models with local attention to support long audios. It can be done later if you see useful.